okio/src/nativeMain/kotlin/okio/Deflater.kt

```diff
@@ -66,47 +70,61 @@ internal class Deflater : DataProcessor() {
   /** Probably [Z_NO_FLUSH], [Z_FINISH], or [Z_SYNC_FLUSH]. */
   var flush: Int = Z_NO_FLUSH
 
-  override fun process(): Boolean {
-    check(!closed) { "closed" }
-    require(0 <= sourcePos && sourcePos <= sourceLimit && sourceLimit <= source.size)
-    require(0 <= targetPos && targetPos <= targetLimit && targetLimit <= target.size)
+  actual constructor() : this(Z_DEFAULT_COMPRESSION, false)
 
-    source.usePinned { pinnedSource ->
-      target.usePinned { pinnedTarget ->
-        val sourceByteCount = sourceLimit - sourcePos
-        zStream.next_in = when {
-          sourceByteCount > 0 -> pinnedSource.addressOf(sourcePos) as CPointer<UByteVar>
-          else -> null
-        }
-        zStream.avail_in = sourceByteCount.toUInt()
+  internal val dataProcessor: DataProcessor = object : DataProcessor() {
```

Comment:
Super annoying; I can’t extend `DataProcessor` in Kotlin/Native because that doesn’t match the inheritance hierarchy in Kotlin/JVM.